### ISSUE TITLE
Field mapping is invalid: OPRID => Alias, PS_OPRID__c

### ISSUE THREAD

**Issue by tggagne** [forcedotcom/dataloader#76]
Inside LoadMapper, it seems to think the sfdc field name is "Alias, PS_OPRID__c," instead of there being two separate fields, Alias and PS_OPRID__c, each mapped to the same source field, OPRID.

Curiously, this works for literal mappings like...

```
"en_US"=LocaleSidKey, LanguageLocaleKey
```

But not

```
OPRID=Alias, PS_OPRID__c
```

**Comment by tggagne** [comment]
Why should it matter whether the mapping is a constant or not?

(from Mapper.java)
```
    @Override
    protected void putPropertyEntry(Entry<Object, Object> entry) {
        String dao = (String)entry.getKey();
        String sfdc = (String)entry.getValue();
        if (isConstant(dao))
            putConstant(sfdc, dao);
        else if (!hasDaoColumns() || hasDaoColumn(dao)) putMapping(dao, sfdc);
    }
```
The problem could be that the map is built from with the source value as the key and the destination value as the value, which requires the source values be unique, or at least implies it.

Better if the destination values (sfdc fields) were the key because it would be incorrect for a single field to be the destination of two sources.  It is more likely (and useful) if a single source column can be mapped to multiple destinations fields.

**Comment by tggagne** [comment]
So LoadMapper simply creates a map, leaving (in my case) oprid => "alias, operid__c."  

To get past the AbstractLoadAction.verifyMappingAreValid() we'll need that function to parse the value looking for multiple destinations, then we'll need whatever function that actually maps the data from the source to the destination to set the value of each of the destination fields.

Does that make sense to anyone else?  Does it seem a reasonable plan?

**Comment by tggagne** [comment]
PartnerClient.getField(String apiName) get's a partner field, which is not a string, and sets it as the value of "oprid,", which means the map's value may have to be a list of com.sforce.soap.partner.Field.  I wonder how obnoxious that may be to change...

Does anyone know where the data mappings actually take place? I mean, when the source column's value is copied to the destination field before insert/upsert?

**Comment by tggagne** [comment]
It looks like LoadMapper.mapData() could be modified to handle a list<Field> so that when it creates the new Row() it could map multiple values.  Perhaps LoadMapper.getMapping() should return the list<Field>.  It should be coded to always return a list, empty or not.  That way callers could specifically look for empty lists to detect mapping errors, or iterate through the list assigning the values.

I think I may try that.

**Comment by ashitsalesforce** [comment]
Looks like the issue is resolved with PR #78. Closing it.
